src/libraries/System.Formats.Nrbf/src/System/Formats/Nrbf/ArrayRecord.cs

```diff
@@ -46,6 +52,11 @@ private protected ArrayRecord(ArrayInfo arrayInfo)
 
     private protected ArrayInfo ArrayInfo { get; }
 
+    internal bool IsJagged
+        => ArrayInfo.ArrayType == BinaryArrayType.Jagged
+        // It is possible to have binary array records have an element type of array without being marked as jagged.
+        || TypeName.GetElementType().IsArray;
```

Comment:
@JeremyKuhne this handles the scenario you have mentioned offline (there is also a test for that). Thank you again for pointing this out!